prow/cmd/branchprotector/protect.go

```diff
@@ -143,7 +143,7 @@ func main() {
 type client interface {
 	RemoveBranchProtection(org, repo, branch string) error
 	UpdateBranchProtection(org, repo, branch string, config github.BranchProtectionRequest) error
-	GetBranches(org, repo string) ([]github.Branch, error)
+	GetBranches(org, repo string, onlyProtected bool) ([]github.Branch, error)
```

Comment:
Does this need to return a []Branch ? []string would suffice.

Reply:
Can you drop the protected bool then, since it is not always set.

Reply:
Are you suggesting removing it from the `type Branch struct`?

Given that [GetBranch](https://developer.github.com/v3/repos/branches/#get-branch) and [ListBranches](https://developer.github.com/v3/repos/branches/#list-branches) both include this value I think it is useful to have the client parse this value when it is present.

Theoretically it could be a pointer so that we can distinguish unset and false, if that is interesting to you? I'm not sure it is important though.